app/controllers/api/v2/templates_controller.rb

```diff
@@ -0,0 +1,27 @@
+module Api
+  module V2
+    class TemplatesController < ::Api::V2::BaseController
+
+    api :GET, "/templates/sync/", N_("Initiate Sync")
+    param :repo, String, :required => false, :desc => N_("Sync templates from a different repo.")
+    param :branch, String, :required => false, :desc => N_("Branch in Git repo.")
+    param :prefix, String, :required => false, :desc => N_("The string all imported templates should begin with.")
+    param :dirname, String, :required => false, :desc => N_("The directory within the git tree containing the templates.")
+    param :filter, String, :required => false, :desc => N_("Import names matching this regex (case-insensitive; snippets are not filtered).")
+    param :associate, String, :required => false, :desc => N_("Associate to OS's, Locations & Organizations. Options are: always, new or never.")
+    def sync
+      results = ForemanTemplates::TemplateImporter.new({
+        verbose:   params['verbose'],
```

Comment:
please avoid the white-space aligning, it's hard to review changes when new longer key is added